### ISSUE TITLE
PS Class type not found error with using module

### ISSUE THREAD

**Issue by ThomasNieto** [PowerShell/vscode-powershell#2887]
### System Details

System Details Output
```
### VSCode version: 1.47.3 91899dcef7b8110878ea59626991a18c8a6a1b3e x64

### VSCode extensions:
Damien.autoit@1.0.5
DavidAnson.vscode-markdownlint@0.36.2
eamodio.gitlens@10.2.2
fernandoescolar.vscode-solution-explorer@0.3.10
GrapeCity.gc-excelviewer@3.0.39
jmrog.vscode-nuget-package-manager@1.1.6
mechatroner.rainbow-csv@1.7.1
mhutchie.git-graph@1.25.0
ms-dotnettools.csharp@1.22.1
ms-mssql.mssql@1.9.0
ms-python.python@2020.7.96456
ms-vscode-remote.remote-wsl@0.44.4
ms-vscode.cpptools@0.29.0
ms-vscode.powershell@2020.6.0
ms-vscode.powershell-preview@2020.7.0
redhat.vscode-xml@0.13.0
streetsidesoftware.code-spell-checker@1.9.0
zhuangtongfa.material-theme@3.8.5


### PSES version: 2.3.0.0

### PowerShell version:

Name                           Value
----                           -----
PSVersion                      7.1.0-preview.5
PSEdition                      Core
GitCommitId                    7.1.0-preview.5
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Issue Description

PS Class with using module statement throws type not found. If in the integrated PS console I run `using module X` and then remove and add back the using statements the error will away.

It looks as though VS code is not running the using statements while editing the script.

#### Expected Behaviour

Open a class file with types residing in a module and no type not found error is thrown.

#### Actual Behaviour

Type not found until manually running using statements in integrated console.

### Attached Logs

Follow the instructions in the [troubleshooting docs](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#logs)
about capturing and sending logs.

**Comment by SeeminglyScience** [comment]
@ThomasNieto can you provide an example?

`using module` statements are notoriously finicky when it comes to static analysis.

**Comment by ThomasNieto** [comment]
@SeeminglyScience Here is a way to reproduce:

I have a classtest.dll which is the `RootModule` in the `Test` module.

```csharp
namespace ClassTest.Test
{
    public class Animal
    {
        public string Name { get; set; }
    }
}

```

The animal class will throw a type not found error. To get it to work I have to do the following:

1. In integrated console run: `using module Test`
2. Cut and paste either using statement.
3. Animal type no longer has the error

This workflow can be reproduced over and over by restarting the integrated console.

```powershell
using module Test
using namespace ClassTest.Test

class Dog : Animal
{

}`

**Comment by SeeminglyScience** [comment]
Yeah that's one of the situations where it doesn't work, binary modules.  Here's the issue tracking it: PowerShell/PowerShell#6653 (sort of, that needs to come first but even then there's a lot of obstacles to jump from `using assembly` to `using module` for static analysis of assemblies)

**Comment by ThomasNieto** [comment]
I get having to run using module in the console to load the assemblies but why does it require redefining the using statements in the script?

**Comment by SeeminglyScience** [comment]
You sure it isn't just any edit? PSSA (where those warnings come from kinda sorta) wouldn't be ran again until there's an edit of some sort.

**Comment by ThomasNieto** [comment]
Yeah, any edit seems to do.

**Comment by SydneyhSmith** [comment]
There isn't much we can do in the PowerShell extension, something that would be nice to have is re-running diagnostics when a module is loaded, but PowerShell doesnt have an event for that and likely wont in the intermediate future leaving us without many options

**Comment by JustinGrote** [comment]
@andschwa @SeeminglyScience  there currently is a situation where we get both the "unable to find type" and the PSScriptAnalyzer "Ignoring TypeNotFound" info. Maybe if we detect both of these we can silently suppress the error Problem and only present the information warning?

![image](https://user-images.githubusercontent.com/15258962/144144280-bde2a713-38c4-4154-ac42-029c7db4978d.png)

If so it may be worth reopening this issue to address.

**Comment by andschwa** [comment]
Just FYI @JustinGrote I'm adding you to the bot's ignore list for it's auto responder lol. Setting it to exclude "collaborators" and up now.

To the issue at hand, I'm not sure! But I can reopen for us to look at again.
